client/src/components/Today/RadioOption.tsx

```diff
@@ -0,0 +1,56 @@
+import { useTheme } from "@emotion/react";
+import {
+	Radio,
+	type RadioCardProps,
+	RadioIndicator,
+	Tooltip,
+} from "@mantine/core";
+import type { LucideIcon } from "lucide-react";
+import type { MainTheme } from "@/lib/style/theme";
+import { colors } from "@/lib/theme/colors";
+import { spacingValue } from "@/lib/theme/snippets/spacing";
+
+type RadioOptionProps = RadioCardProps & {
+	Icon: LucideIcon;
+	tooltipLabel: string;
+};
+
+/** Small generalization of RadioCard for specific use cases.
+ * @usage use this in a user setting that has multiple options (like the habit
+ *    completion filter preference).
+ * @usage use this in the filters for habits and tasks for completion state.
+ */
+export default function RadioOption({
+	Icon,
+	tooltipLabel,
+	checked,
+	...props
+}: RadioOptionProps) {
+	const theme = useTheme() as MainTheme;
+
+	return (
+		<Tooltip
+			label={tooltipLabel}
+			events={{ hover: true, focus: true, touch: true }}
+		>
+			<Radio.Card
+				{...props}
+				checked={checked}
+				styles={{
+					card: {
+						display: "flex",
+						alignItems: "center",
+						borderRadius: "50%",
+						width: "max-content",
+						color: theme.colors.text.main[0],
+						padding: spacingValue.smaller,
+						backgroundColor: checked ? colors.green.secondary : "var(--bg-1-2)",
+					},
+				}}
+			>
+				<RadioIndicator hidden />
+				<Icon size={16} color={checked ? colors.dark[0] : "currentColor"} />
```

Comment:
This color should be tied to the card style (for contrast reasons), so if we export the card style from the theme lib, then this should be defined right alongside it for clarity.